xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/script/RefactoringScriptService.java

```diff
@@ -661,4 +662,44 @@ private WikiReference getCurrentWikiReference()
 
         return result;
     }
+
+    /**
+     * Creates a request to permanently delete a specified batch of deleted documents from the recycle bin.
+     *
+     * @param batchId the ID of the batch of deleted documents to permanently delete
+     * @return the permanently delete request
+     * @since 10.10RC1
+     */
+    public PermanentlyDeleteRequest createPermanentlyDeleteRequest(String batchId)
```

Comment:
It's missing a method to actually use that PermanentlyDeleteRequest.

Reply:
The request is actually used by calling a `jobExecutor.execute(type, request)`. But maybe you mean that I should add a method in the RefactoringScriptService to call the executor, like what is done for copy/move/etc? I didn't do it because it wasn't done for restore. I might add it for both of them, if you think there's a value for that.

Reply:
> The request is actually used by calling a jobExecutor.execute(type, request)

You are talking about a single use case here.

The problem here is that what  createPermanentlyDeleteRequest returns cannot be used from (Velocity) script point of view so this method does not make any sense in a script service. It's also not consistent with the other jobs, and yes this was a mistake in restore case too.

Reply:
ok. Do you think I should make it private and only expose the method I'll create which will also run the request? And same question for restore?

Reply:
In theory restore and permanentlyDelete should expose the same kind of API than the other jobs:
* method(s) to create request from some values
* method to run the job with the passed request
* method(s) to rune the job with passed values

Reply:
I made some changes in that way. As usual the PR is growing, maybe it worth it to make a separate PR for the refactorings of xwiki-platform-refactoring, WDYT?

Reply:
> the refactorings of xwiki-platform-refactoring

Not sure what you are referring to, the methods you added to the script services looks OK.

Reply:
Added.